pkg/providers/ovirt/templates/template-finder_test.go

```diff
@@ -63,6 +64,28 @@ var _ = Describe("Finding a Template", func() {
 		Expect(err).To(Not(BeNil()))
 		Expect(template).To(BeNil())
 	})
+	It("should find newer template:", func() {
+		now := metav1.Now()
+		newer := metav1.NewTime(now.Add(time.Duration(1 * time.Minute)))
+		findTemplatesMock = func(name *string, os *string, workload *string, flavor *string) (*templatev1.TemplateList, error) {
+			template1 := createTemplate(name, os, workload, flavor)
+			template1.CreationTimestamp = now
+
+			template2 := createTemplate(name, os, workload, flavor)
+			template2.CreationTimestamp = newer
+
+			templateList := createTemplatesList(template1, template2)
+			return templateList, nil
+		}
+
+		vmOS := ovirtsdk.OperatingSystem{}
+		vmOS.SetType("rhel")
+		vm := ovirtsdk.NewVmBuilder().Os(&vmOS).MustBuild()
+		template, err := templateFinder.FindTemplate(vm)
+
+		Expect(err).To(BeNil())
+		Expect(newer.Equal(&template.CreationTimestamp)).To(BeTrue())
```

Comment:
Is there a reason for this assertion not to be
```go
Expect(template.CreationTimestamp).To(Equal(newer))
```?

Reply:
This `time` type implements its own `Equal` method.

Reply:
I may be nitpicking but to me it looks rather unusual and by it suggests that some extra measures need to be applied to verify the behaviour. 

Are the special features of `time.Time.Equal` really needed? As per documentation it allows for two "times" to be equal if they are in different locations but represent the same instant - likely in contrast to the `reflect.DeepEqual` that is used by the matcher. Do we, from the client perspective, allow the code under test to change the location of the time?

Reply:
I think one of the benefit for using k8s type `Equal` is handling `nil`. For this specific test we know both timestamps won't be `nil` but it feels right to use `Equal` defined for the type.

Reply:
If you feel we should use standard matcher I am ok with it.

Reply:
The reasoning behind using the dedicated `Equal` absolutely make sense. My reluctance towards using it is caused by the sheer surprise of seeing an uncommon assertion construction that made me stop and investigate why it looks this way. When the code changes in the future someone will probably need to investigate it again.
I don't want to push it, though.

Reply:
Thanks for the change:)

Reply:
Too late. I have updated the PR :stuck_out_tongue: